sos/plugins/openstack_nova.py

```diff
@@ -28,44 +29,37 @@ class OpenStackNova(Plugin):
-                                        " which is required" % (os_var))
-            self.add_cmd_output(
-                "nova service-list",
-                suggest_filename="nova_service_list")
-            self.add_cmd_output(
-                "nova-manage db version",
-                suggest_filename="nova_db_version")
-            self.add_cmd_output(
-                "nova-manage fixed list",
-                suggest_filename="nova_fixed_ip_list")
-            self.add_cmd_output(
-                "nova-manage floating list",
-                suggest_filename="nova_floating_ip_list")
-            self.add_cmd_output(
-                "nova flavor-list",
-                suggest_filename="nova_flavor_list")
-            self.add_cmd_output(
-                "nova network-list",
-                suggest_filename="nova_network_list")
-            self.add_cmd_output(
-                "nova list",
-                suggest_filename="nova_vm_list")
-            self.add_cmd_output(
-                "nova agent-list",
-                suggest_filename="nova_agent_list")
-            self.add_cmd_output(
-                "nova version-list",
-                suggest_filename="nova_version_list")
-            self.add_cmd_output(
-                "nova host-list",
-                suggest_filename="nova_host_list")
+        # commands we do not need to source the environment file
+        self.add_cmd_output("nova-manage db version")
+        self.add_cmd_output("nova-manage fixed list")
+        self.add_cmd_output("nova-manage floating list")
+
+        vars = [p in os.environ for p in [
+                'OS_USERNAME', 'OS_PASSWORD', 'OS_TENANT_NAME']]
+        if not all(vars):
+            self.soslog.warning("Not all environment variables set. Source "
+                                "the environment file for the user intended "
+                                "to connect to the OpenStack environment.")
+        else:
+            self.add_cmd_output("nova service-list")
+            self.add_cmd_output("openstack flavor list --long")
+            self.add_cmd_output("nova network-list")
+            self.add_cmd_output("nova list")
+            self.add_cmd_output("nova agent-list")
+            self.add_cmd_output("nova version-list")
+            self.add_cmd_output("nova host-list")
+            self.add_cmd_output("openstack quota show")
+            self.add_cmd_output("openstack hypervisor stats show")
+            # get details for each nova instance
+            cmd = "openstack server list -f value"
+            nova_instances = self.call_ext_prog(cmd)['output']
+            for instance in nova_instances.splitlines():
+                instance = instance.split()[0]
+                cmd = "openstack server show %s" % (instance)
+                self.add_cmd_output(
+                    cmd,
+                    suggest_filename="instance-" + instance + ".log")
```

Comment:
Ditto these changes - would have been much better to do the code shuffling in a separate patch.